apps/site/assets/ts/app/channels.ts

```diff
@@ -0,0 +1,104 @@
+import { Socket, Channel } from "phoenix";
+
+declare global {
+  interface Window {
+    channels: { [id: string]: Channel };
+    socket: Socket;
+  }
+}
+
+type UpdateEventName = "reset" | "add" | "update";
+
+interface UpdateEvent<DataType> {
+  event: UpdateEventName;
+  data: DataType;
+}
+
+interface RemoveEvent {
+  event: "remove";
+  data: string[];
+}
+
+export type SocketEvent<DataType> = UpdateEvent<DataType> | RemoveEvent;
+
+const isVehicleChannel = (channelId: string): boolean =>
+  channelId.includes("vehicles:") && channelId !== "vehicles:remove";
+
+const joinChannel = <T>(channelId: string): void => {
+  if (!window.socket) return;
+
+  if (!window.channels[channelId]) {
```

Comment:
question: This array doesn't need to be initialised because we are using string keys to access it making it like a dict?